esp-hal-common/src/gpio.rs

```diff
@@ -1062,54 +1139,60 @@ where
         });
     }
 
-    pub fn into_push_pull_output(self) -> GpioPin<Output<PushPull>, RA, PINTYPE, GPIONUM> {
+    pub fn into_push_pull_output(self) -> GpioPin<Output<PushPull>, RA, IRA, PINTYPE, GPIONUM> {
         self.init_output(GPIO_FUNCTION, false);
         GpioPin {
             _mode: PhantomData,
             _pintype: PhantomData,
             reg_access: self.reg_access,
+            ira: self.ira,
```

Comment:
Is the ira field needed? Looks like the usage doesn't require self so it can just be called with `IRA::XXXX`? IRA should probably be a `PhantomData` field, unless I'm missing something?

Reply:
I think `PhantomData` should be enough - also for `reg_access` ... I assumed it's zero-sized in both cases but anyway probably better to stick to `PhantomData` ... will check

Reply:
ah I see `BankGpioRegisterAccess` trait functions take `&self` but for no reason